nix/package.nix

```diff
@@ -0,0 +1,54 @@
+{ stdenv, elixir, rebar3, hex, callPackage, git, cacert }:
+
+let
+  fetchMixDeps = callPackage ./fetch-mix-deps.nix {};
+in stdenv.mkDerivation rec {
+  name = "elixir-ls";
+  version = "dev";
+
+  nativeBuildInputs = [ elixir hex git deps cacert ];
+
+  deps = fetchMixDeps {
+    name = "${name}-${version}";
+    inherit src;
+  };
+
+  src = ../.;
```

Comment:
We should probably use `lib.cleanSource`(`With`) here to avoid triggering rebuilds just because some autosave file changed, and to keep the input clean in general.

I'd probably prefer some specialised `lib.cleanSourceWith` call, which filters `.git/`, `.git*`, `deps/`, `_build/`, `.vscode/`, `#*#`, `*~`, `result`(!) just to avoid artifacts or most common editor files slipping in (Please extend the list in case I've missed some other editors backup/lock-files).

Reply:
I wasn't aware of such helper function. You are probably right that it should be used there.